src/backlinks/handler.js

```diff
@@ -139,13 +139,50 @@ export default async function auditBrokenBacklinks(message, context) {
       auditResult,
     };
 
-    await dataAccess.addAudit(auditData);
+    const audit = await dataAccess.addAudit(auditData);
     const data = {
       type,
       url: site.getBaseURL(),
       auditContext,
       auditResult,
     };
+
+    const opportunities = await dataAccess.Opportunity.allBySiteIdAndStatus(siteId, 'NEW');
+    let brokenBacklinksOppty = opportunities.find((oppty) => oppty.getType() === 'broken-backlinks');
+
+    if (!brokenBacklinksOppty) {
+      const opportunityData = {
+        siteId: site.getId(),
+        auditId: audit.getId(),
+        runbook: 'https://adobe.sharepoint.com/:w:/r/sites/aemsites-engineering/_layouts/15/doc2.aspx?sourcedoc=%7BAC174971-BA97-44A9-9560-90BE6C7CF789%7D&file=Experience_Success_Studio_Broken_Backlinks_Runbook.docx&action=default&mobileredirect=true',
+        type: 'broken-backlinks',
+        origin: 'AUTOMATION',
+        title: 'Authoritative Domains are linking to invalid URLs. This could impact your SEO.',
+        description: 'Provide the correct target URL that each of the broken backlinks should be redirected to.',
+        guidance: {
+          steps: [
+            'Review the list of broken target URLs and the suggested redirects.',
+            'Manually override redirect URLs as needed.',
+            'Copy redirects.',
+            'Paste new entries in your website redirects file.',
+            'Publish the changes.',
+          ],
+        },
+        tags: ['traffic-acquisition'],
+      };
+      brokenBacklinksOppty = await dataAccess.Opportunity.create(opportunityData);
+    }
+
+    const suggestions = await brokenBacklinksOppty.addSuggestions(data.auditResult.brokenBacklinks);
```

Comment:
@dzehnder What about the suggestions that are already existing?

**Example:**
- On 1st Nov. an audit runs and detects 3 broken links: A, B, C and this code creates 3 entries in the Suggestions table.
- On 2nd Nov, a back-office-ui user adds a new broken-backlink D to suggestions table.
- On 3rd Nov, a ESS user marks the backlink B as Ignored.
- On 7th Nov, another audit runs and detects 4 broken backlinks: A', B', E, F

**What is the expected state of Suggestions table, after the audit runs on 7th Nov?**

- A, B(ignored), C, D, A', B', E, F
- A, B(ignored), C, D, E, F
- A', B', C, D, E, F
- A', B', E, F
- A', B'(ignored), E, F
- something else?

Reply:
if a backlink was marked as ignored it should not appear anymore in the ESS UI and stay in the ignored list. But that is handled as part of the site configuration. 
So the state of the suggestions table does not include the ignored property, it only has the data of the current active broken backlinks.
In that case it would look like this:

- A, B, C, D, E, F

Reply:
Also started a thread on slack to clarify the expectations with the PM: https://cq-dev.slack.com/archives/C05A45JBP9N/p1732545991035499

Reply:
Conclusion from slack:

> A', B' (ignored), D, E, F

seems correct.

PR needs to be updated accordingly

Reply:
done! I also wrote a helper function that everyone can use for their audits, to not have to think about this too much